packages/api/models/executions.js

```diff
@@ -76,7 +83,15 @@ class Execution extends Manager {
     const concurrencyLimit = process.env.CONCURRENCY || 10;
     const limit = pLimit(concurrencyLimit);
 
-    const updatePromises = oldExecutionRows.Items.map((row) => limit(() => this.update({ arn: row.arn }, {}, ['originalPayload', 'finalPayload'])));
+    const updatePromises = oldExecutionRows.Items.map((row) => limit(() => {
+      if (row.status === 'completed' && !disableComplete && row.updatedAt <= completeMaxMs) {
+        return this.update({ arn: row.arn }, {}, ['originalPayload', 'finalPayload']);
+      }
+      if (!(row.status === 'completed') && !disableNonComplete && row.updatedAt <= nonCompleteMaxMs) {
```

Comment:
Real nit-picky but it might be quicker to short-circuit these conditionals by starting with `!disableNonComplete`

Reply:
I love constructive nit-picks, often they make the code (or me.... or both!)  better. 

In this case, 💯 .     

e7c9d8cd